src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -630,25 +634,46 @@ internal PhysicalLocation CreatePhysicalLocation(PhysicalLocationVersionOne v1Ph
                 physicalLocation = new PhysicalLocation
                 {
                     FileLocation = CreateFileLocation(v1PhysicalLocation),
-                    Region = CreateRegion(v1PhysicalLocation.Region)
+                    Region = CreateRegion(v1PhysicalLocation)
                 };
             }
 
             return physicalLocation;
         }
 
-        internal Region CreateRegion(RegionVersionOne v1Region)
+        internal Region CreateRegion(PhysicalLocationVersionOne v1PhysicalLocation)
         {
             Region region = null;
 
-            if (v1Region != null)
+            if (v1PhysicalLocation?.Region != null)
             {
-                region = CreateRegion(v1Region.StartColumn,
-                                      v1Region.StartLine,
-                                      v1Region.EndColumn,
-                                      v1Region.EndLine,
-                                      v1Region.Length,
-                                      v1Region.Offset);
+                RegionVersionOne v1Region = v1PhysicalLocation.Region;
+
+                region = new Region
+                {
+                    ByteLength = v1Region.Length,
+                    ByteOffset = v1Region.Offset,
+                    EndColumn = v1Region.EndColumn,
+                    EndLine = v1Region.EndLine,
+                    StartColumn = v1Region.StartColumn,
+                    StartLine = v1Region.StartLine
+                };
+
+                bool startIsTextBased = v1Region.StartLine > 0;
+                bool endIsTextBased = v1Region.EndLine > 0 || v1Region.EndColumn > 0;
+
+                if (startIsTextBased && endIsTextBased && v1Region.EndColumn == 0)
+                {
+                    region.EndColumn = v1Region.StartColumn;
+                }
+                else if (startIsTextBased)
+                {
+                    region.CharLength = GetRegionCharLength(v1Region, v1PhysicalLocation.Uri);
+                }
+                else
```

Comment:
This was a bug in my sample code. If we get here, `startIsTextBased` is false. If `endIsTextBased` is also false, then the entire region was specified by a byte offset and a byte length, and we've already copied those properties. So the `else` should be `else if (endIsTextBased)`.

Reply:
Fixed

Reply:
Fixed